src/System.Private.Xml/src/System/Xml/Core/QueryOutputWriterV1.cs

```diff
@@ -239,10 +239,12 @@ public override void WriteChars(char[] buffer, int index, int count)
 
         public override void WriteBase64(byte[] buffer, int index, int count)
         {
-            if (!_inAttr && (_inCDataSection || StartCDataSection()))
-                _wrapped.WriteBase64(buffer, index, count);
-            else
-                _wrapped.WriteBase64(buffer, index, count);
+            if (!_inAttr && !_inCDataSection)
+            {
+                StartCDataSection();
```

Comment:
I'm actually wondering if it should be called at all since we don't actually start CDATA section here. I suspect this was a copy&paste fail from `WriteString`

Reply:
Thanks.  Is there a test I should add to go with it?  And remove the whole if block?

Reply:
It might be rather hard to instantiate this specific writer through public APIs since seems the only way to get there is through https://github.com/dotnet/corefx/blob/master/src/System.Private.Xml/src/System/Xml/Core/XmlWriterSettings.cs#L882 and also several other conditions. I'd just leave this PR as is or create it through reflection

Reply:
ok, I just double checked and it seem it doesn't open extra CDATA

```
        static XmlWriter CreateQueryOutputWriterV1(XmlWriter writer)
        {
            string typeName = "System.Xml.QueryOutputWriterV1";
            Type t = typeof(XmlWriter).Assembly.GetType(typeName);
            return (XmlWriter)Activator.CreateInstance(t, writer, writer.Settings);
        }
        static void Main(string[] args)
        {

            using (XmlWriter writer = CreateQueryOutputWriterV1(XmlWriter.Create(Console.Out, new XmlWriterSettings() {  })))
            {
                Console.WriteLine(writer.GetType().FullName);
                writer.WriteStartDocument();
                writer.WriteStartElement("root");
                writer.WriteBase64(new byte[] { 1, 2, 3 }, 0, 3);
            }
        }
```

Reply:
Ok, thanks.  I'll just go with the transformation then, and we could later remove it if doing so proves fruitful.